### DISCUSSION TITLE
remaping V<<Esc> throws errors

### DISCUSSION THREAD

**Discussion by sourcevault** [NeoVintageous/NeoVintageous#978] [General]
https://github.com/NeoVintageous/NeoVintageous/assets/20347647/c0220f56-c7a7-479a-b2bd-18a8ddb022a7

These two key remap throwing strange errors:

```vim.rc
nnoremap <leader>i<left> V :Unindent<CR><Esc>
nnoremap <leader>i<left> V<<Esc>
```

**Comment by sourcevault** [comment]
question about remapping combined commands in vim like "ciw" when remapping "w".

https://github.com/NeoVintageous/NeoVintageous/assets/20347647/3214d99b-b4bd-44fb-b5ec-c879ee0c3136

**Reply by gerardroche** [reply]
Yes. Recursive mappings are not supported. 

Recursive mappings, i.e., `*map` mappings, means `ci<Right>`:

1. `c` and then check if c has been remapped, and then recursively find the rhs (right hand side of the mapping)
2. `i` and then check if i has been remapped, and then recursively find the rhs (right hand side of the mapping)
3. `<Right>` is remapped to w. 

In non recursive mappings, i.e., the `*noremap` mappings, `<Right>` resolves to the default.

Recursive mappings were not implemented because they were much more difficult to implement, e.g., they can cause infinite loops.

**Comment by gerardroche** [comment]
I think you need to use `<lt>` for `<` because it's a special character. Vim might use a heuristic to tell if you meant literal `<` or key `<`, but it won't always know, so you can use `<lt>` to always mean key.

**Reply by sourcevault** [reply]
Thanks for that. 

The indent and unindent command is really important and useful for python and LiveScript.

**Reply by gerardroche** [reply]
Some indent commands 

Reindent: `==`

Indent: `>>`

Indent 3 lines `3>>`

Indent from cursor to end of paragraph: `>}`

Indent indentation level including outer lines (text object): `>iI`

Indent to end of file: `>G` Like commenting to eof `gcG`

Deindent `<<`

**Reply by sourcevault** [reply]
That is really cool, especially the `<<` and `>>`.

**Comment by gerardroche** [comment]
I'm surprised your mappings work as well as they seem to work tbh. Your level of remapping is extensive. It's far from the regular use case.
